### DISCUSSION TITLE
Metric Targets for Monitoring and Scaling

### DISCUSSION THREAD

**Discussion by akochpaycor** [dotnet/yarp#2204] [Q&A] [answered]
Based on reading the documentation and other Q&A discussions, it looks like it is possible for YARP to be both CPU or Memory bound.

Do you have any rule-of-thumb thresholds for CPU or Memory that can be used for 1) Scaling Rules on a hosting App Service Plan 2) Warning and Critical level monitoring alerts? (ex. CPU over 90%, Memory over 80% for > 5 minutes)

Are there any other App Service Plan standard metrics that you believe are strongly correlated to YARP health? (ex. Http Queue Length)

**Comment by MihaZupan** [comment, marked as answer]
I don't believe we have such guidance available, but running YARP should be similar to running any other ASP.NET Core server application.

> Are there any other App Service Plan standard metrics that you believe are strongly correlated to YARP health? (ex. Http Queue Length)

If you are exporting .NET metrics for `System.Net.Http`, counters like `http11-requests-queue-duration` and `http20-requests-queue-duration` can be helpful to spot overloaded connections to the backend destinations.
